src/software/ai/hl/stp/play/play.h

```diff
@@ -66,6 +74,32 @@ class Play
      */
     virtual bool invariantHolds(const World& world) const = 0;
 
+    /**
+     * Returns true if the Play is done and false otherwise. The Play is considered
+     * done when its coroutine is done (the getNextTactics() function has no
+     * more work to do)
+     *
+     * @return true if the Play is done and false otherwise
+     */
+    bool done() const;
+
+    /**
+     * Gets Intents from the Play given the assignment algorithm and world
+     *
+     * @param robot_to_tactic_assignment_algorithm The algorithm for assigning robots to
+     * tactics
+     * @param motion_constraint_builder Builds motion constraints from tactics
+     * @param world The updated world
+     *
+     * @return the vector of intents to execute
+     */
+    std::vector<std::unique_ptr<Intent>> get(
```

Comment:
nit; but `getIntents` might be a bit more descriptive if a user is just reading a call to this function without the jdoc in front of them.

Reply:
shouldn't this be obvious from the function definition: `std::vector<std::unique_ptr<Intent>> get`

Reply:
this will end up confusing if someone assigns this to `auto`. `play.get()` doesn't tell me anything about what this function is supposed to do

Reply:
I would hope people would name variables appropriately, i.e. `auto intents = play.get()`

Reply:
If you're choosing between the two of them, might as well choose the more descriptive one no? If I'm using autocomplete on an instance of this class, I'm definitely going to have to go lookup the function signature to understand what this does.

Reply:
I don't understand the big fuss over this since we will only be calling this from STP.

> If I'm using autocomplete on an instance of this class, I'm definitely going to have to go lookup the function signature to understand what this does.

Reply:
I suppose the difference for me is I have no idea what the function called `get` does, but I at least know the semantics of what I'm getting if it's called `getIntents` (vs. the syntactics, which is what `getIntentUniquePtrVector` conveys). 

I'm not saying `get` is incorrect, I just think `getIntents` is a more semantically descriptive name that's more generally consistent with how getter functions are named in this codebase.

(All the above said, all my comments are really just intended as suggestions. It's absolutely your guys's codebase now, so please don't feel like you have to justify anything to me!)

Reply:
Well, there seem to be other people on the team that aren't convinced, so it seems worth explaining my thinking.
`getIntents` could return so many things: `std::set<std::weak_ptr<Intent>>`? `std::vector<std::shared_ptr<Intent>>`? `std::unorderered_set<Intent*>`? `std::vector<std::shared_ptr<const Intent>>`? 
I don't think we should be making these assumptions about how our codebase works to somehow deduce that `getIntents` means a vector of unique ptrs. Please please please read function signatures.